### PR TITLE
test: add e2e experiment list pagination test

### PR DESCRIPTION
**Pull request by ashtonG**
## Ticket
ET-754

## Description
This adds a pagination test to the e2e experiment list test suite. We check page url parameter after interacting with the pagination element. Because the table rows are virtualized, we cannot directly check the row count based on what is in the html dom.

## Test Plan
React-e2e tests should pass

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/e2e/tests/experimentList.spec.ts [hunk 1]
```diff
@@ -19,11 +19,11 @@ test.describe('Experiment List', () => {
   test.beforeAll(async ({ backgroundAuthedPage, newWorkspace, newProject }) => {
     const projectDetailsPageSetup = new ProjectDetails(backgroundAuthedPage);
     await projectDetailsPageSetup.gotoProject(newProject.response.project.id);
-    await test.step('Create an experiment', async () => {
+    await test.step('Create experiments', async () => {
       await expect(
         projectDetailsPageSetup.f_experimentList.tableActionBar.count.pwLocator,
       ).toContainText('experiment');
-      Array(4)
+      Array(52)
```
Comment (ashtonG) [inline webui/react/src/e2e/tests/experimentList.spec.ts:26]: the count here is increased to ensure the page size options show up

Comment (JComins000) [reply webui/react/src/e2e/tests/experimentList.spec.ts:26]: if this isn't necessary for the other tests, i would ask that pagination tests get their own describe block for the other experiments

File: webui/react/src/e2e/tests/experimentList.spec.ts [hunk 2]
```diff
@@ -109,6 +109,14 @@ test.describe('Experiment List', () => {
       await tableFilter.close();
       await waitTableStable();
     });
+    await test.step('Ensure pagination options', async () => {
```
Comment (JComins000) [inline webui/react/src/e2e/tests/experimentList.spec.ts:112]: this is good.

File: webui/react/src/e2e/tests/experimentList.spec.ts [hunk 3]
```diff
@@ -393,4 +401,20 @@ test.describe('Experiment List', () => {
       }).toPass();
     });
   });
+  test('Pagination', async () => {
+    const expectPageNumber = async (pageParam: string | null) => {
+      await projectDetailsPage._page.waitForResponse((res) => {
+        return res.url().endsWith('experiments-search');
+      });
+      const params = new URL(projectDetailsPage._page.url()).searchParams;
+      expect(params.get('page')).toBe(pageParam);
```
Comment (JComins000) [inline webui/react/src/e2e/tests/experimentList.spec.ts:410]: cool

**Review by JComins000 — APPROVED** [review]
just one comment about keeping the environment cleaner for the non-pagination tests. it's your spec, so it's up to you

File: webui/react/src/e2e/tests/experimentList.spec.ts (hunk 1 above)
Comment (ashtonG) [reply webui/react/src/e2e/tests/experimentList.spec.ts:26]: given that the project will be shared between the describe blocks (the project fixture has to be scoped to the worker), i think separating these blocks will only serve to make this test suite take longer.

is that not true?

Comment (JComins000) [reply webui/react/src/e2e/tests/experimentList.spec.ts:26]: if it's shared then it's no problem

Comment (JComins000) [reply webui/react/src/e2e/tests/experimentList.spec.ts:26]: you could nest a new describe block inside this one with 48 more. that way if I individually run one of the other tests, I won't get 50 experiments that I don't need

Comment (JComins000) [reply webui/react/src/e2e/tests/experimentList.spec.ts:26]: you could nest a new describe block inside this one with 48 more. that way if I individually run one of the other tests, I won't get 50 experiments that I don't need

Comment (ashtonG) [reply webui/react/src/e2e/tests/experimentList.spec.ts]: Good idea -- i implemented it.

**Pull request merged**